pkg/network/tracer/offsetguess/overlaps.go

```diff
@@ -0,0 +1,129 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package offsetguess
+
+import (
+	"unsafe"
+
+	"golang.org/x/exp/slices"
+)
+
+// offset, size tuple
+type offsetRange struct {
+	Offset uint64
+	Size   uint64
+}
+
+// ranges must be sorted by starting offset
+func skipOverlaps(offset uint64, ranges []offsetRange) (uint64, bool) {
+	overlapped := false
+StartOver:
+	for {
+		for _, r := range ranges {
+			next := r.Offset + r.Size
+			if r.Offset <= offset && offset < next {
+				offset = next
+				overlapped = true
+				continue StartOver
```

Comment:
Is this `break StartOver`?

Reply:
No. That would break out of both loops. I want to do the inner loop completely again.

Reply:
You need to pass all the ranges without an overlap to return from this function.